### ISSUE TITLE
Qwen3.5-397b-a17b export fp8, model size extremely small

### ISSUE THREAD

**Issue by tnlin** [modelscope/ms-swift#9325]
### Checklist / 检查清单

- [x] I have searched existing issues, and this is a new bug report. / 我已经搜索过现有的 issues，确认这是一个新的 bug report。

### Bug Description / Bug 描述

输出的397b-moe模型fp8权重只有11gb左右，预期应该是400gb左右，很奇怪
已经有手动加上这个补丁 https://github.com/modelscope/ms-swift/pull/8484


### How to Reproduce / 如何复现

export MODEL=./Qwen3.5-397B-A17B-finetuned
export TARGET=$MODEL-FP8
swift export \
    --model $MODEL \
    --quant_method fp8 \
    --output_dir $TARGET


[INFO:swift] Successfully registered `/usr/local/ms-swift-inner-v4.0/swift/dataset/data/dataset_info.json`.
[INFO:swift] Successfully loaded ./Qwen3.5-397B-A17B-finetuned/args.json.
[INFO:swift] rank: -1, local_rank: -1, world_size: 1, local_world_size: 1
`torch_dtype` is deprecated! Use `dtype` instead!
[INFO:swift] Setting args.lazy_tokenize: True
[INFO:swift] args.output_dir: `./Qwen3.5-397B-A17B-finetuned-FP8`
[INFO:swift] args: ExportArguments(use_ray=False, ray_exp_name=None, device_groups=None, model='./Qwen3.5-397B-A17B-finetuned', model_type='qwen3_5_moe', model_revision=None, task_type='causal_lm', torch_dtype=torch.bfloat16, attn_impl=None, experts_impl=None, new_special_tokens=[], num_labels=None, problem_type=None, rope_scaling=None, device_map=None, max_memory={}, max_model_len=None, local_repo_path=None, init_strategy=None, template='qwen3_5', system=None, max_length=262144, truncation_strategy='delete', max_pixels=None, agent_template=None, norm_bbox=None, use_chat_template=True, padding_side='right', padding_free=False, loss_scale='default', sequence_parallel_size=1, template_backend='swift', response_prefix=None, enable_thinking=None, add_non_thinking_prefix=True, dataset=[], val_dataset=[], cached_dataset=[], cached_val_dataset=[], split_dataset_ratio=0.0, data_seed=42, dataset_num_proc=1, load_from_cache_file=False, dataset_shuffle=True, val_dataset_shuffle=False, streaming=False, interleave_prob=None, stopping_strategy='first_exhausted', shuffle_buffer_size=1000, download_mode='reuse_dataset_if_exists', columns={}, strict=False, remove_unused_columns=True, model_name=None, model_author=None, custom_dataset_info=[], quant_method='fp8', quant_bits=None, hqq_axis=None, bnb_4bit_compute_dtype=torch.bfloat16, bnb_4bit_quant_type='nf4', bnb_4bit_use_double_quant=True, bnb_4bit_quant_storage=None, max_new_tokens=None, temperature=None, top_k=None, top_p=None, repetition_penalty=None, num_beams=1, stream=False, stop_words=[], logprobs=False, top_logprobs=None, structured_outputs_regex=None, tuner_backend='peft', tuner_type='full', train_type=None, adapters=[], external_plugins=[], custom_register_path=[], seed=42, model_kwargs={}, load_args=True, load_data_args=False, packing=False, packing_length=None, packing_num_proc=1, lazy_tokenize=True, use_hf=False, hub_token=None, ddp_timeout=18000000, ddp_backend=None, ignore_args_error=False, use_swift_lora=False, merge_lora=False, safe_serialization=True, max_shard_size='5GB', output_dir='./Qwen3.5-397B-A17B-finetuned-FP8', quant_n_samples=256, quant_batch_size=1, group_size=128, to_cached_dataset=False, template_mode='train', to_ollama=False, to_mcore=False, to_hf=False, mcore_model=None, mcore_adapter=None, thread_count=None, test_convert_precision=False, test_convert_dtype=torch.float32, push_to_hub=False, hub_model_id=None, hub_private_repo=False, commit_message='update files', to_peft_format=False, exist_ok=False)
[INFO:swift] Global seed set to 42
[INFO:swift] Start time of running main: 2026-05-13 10:45:26.789412
[INFO:swift] swift.__version__: 4.1.0.dev0
[INFO:swift] Setting torch_dtype: torch.bfloat16
[INFO:swift] model_kwargs: {'device_map': 'auto', 'dtype': torch.bfloat16}
[INFO:swift] Setting max_ratio: 200. You can adjust this hyperparameter through the environment variable: `MAX_RATIO`.
[INFO:swift] Setting frame_factor: 2. You can adjust this hyperparameter through the environment variable: `FRAME_FACTOR`.
[INFO:swift] Setting fps: 2.0. You can adjust this hyperparameter through the environment variable: `FPS`.
[INFO:swift] Setting fps_min_frames: 4. You can adjust this hyperparameter through the environment variable: `FPS_MIN_FRAMES`.
[INFO:swift] Setting fps_max_frames: 768. You can adjust this hyperparameter through the environment variable: `FPS_MAX_FRAMES`.
[INFO:swift] Setting image_max_token_num: 16384. You can adjust this hyperparameter through the environment variable: `IMAGE_MAX_TOKEN_NUM`.
[INFO:swift] Setting image_min_token_num: 4. You can adjust this hyperparameter through the environment variable: `IMAGE_MIN_TOKEN_NUM`.
[INFO:swift] Setting spatial_merge_size: 2. You can adjust this hyperparameter through the environment variable: `SPATIAL_MERGE_SIZE`.
[INFO:swift] Setting video_max_token_num: 768. You can adjust this hyperparameter through the environment variable: `VIDEO_MAX_TOKEN_NUM`.
[INFO:swift] Setting video_min_token_num: 128. You can adjust this hyperparameter through the environment variable: `VIDEO_MIN_TOKEN_NUM`.
[INFO:swift] model_kwargs: {'device_map': 'auto', 'quantization_config': FineGrainedFP8Config(quant_method=<QuantizationMethod.FP8: 'fp8'>), 'dtype': 'auto'}

Loading weights: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1371/1371 [02:16<00:00, 10.02it/s, Materializing param=model.visual.pos_embed.weight]
[INFO:swift] default_system: None                                                                                                          
[INFO:swift] max_length: 262144                                                                                                            
[INFO:swift] response_prefix: '<think>\n'                                                                                                  
[INFO:swift] agent_template: qwen3_5                                                                                                       
[INFO:swift] norm_bbox: norm1000                                                                                                           
[INFO:swift] Setting ROOT_IMAGE_DIR: None. You can adjust this hyperparameter through the environment variable: `ROOT_IMAGE_DIR`.          
[INFO:swift] Setting QWENVL_BBOX_FORMAT: legacy. You can adjust this hyperparameter through the environment variable: `QWENVL_BBOX_FORMAT`. 
[INFO:swift] The ExportArguments will be saved in: ./Qwen3.5-397B-A17B-finetuned-FP8/args.json
Writing model shards: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:30<00:00, 10.09s/it]
[INFO:swift] model: Qwen3_5MoeForConditionalGeneration(                                                                                    
  (model): Qwen3_5MoeModel(                                                                                                                
    (visual): Qwen3_5MoeVisionModel(                                                                                                       
      (patch_embed): Qwen3_5MoeVisionPatchEmbed(                                                                                           
        (proj): Conv3d(3, 1152, kernel_size=(2, 16, 16), stride=(2, 16, 16))                                                               
      )                                                                                                                                    
      (pos_embed): Embedding(2304, 1152)                                                                                                   
      (rotary_pos_emb): Qwen3_5MoeVisionRotaryEmbedding()                                                                                  
      (blocks): ModuleList(                                                                                                                
        (0-26): 27 x Qwen3_5MoeVisionBlock(                                                                                                
          (norm1): LayerNorm((1152,), eps=1e-06, elementwise_affine=True)                                                                  
          (norm2): LayerNorm((1152,), eps=1e-06, elementwise_affine=True)                                                                  
          (attn): Qwen3_5MoeVisionAttention(                                                                                               
            (qkv): Linear(in_features=1152, out_features=3456, bias=True)                                                                  
            (proj): Linear(in_features=1152, out_features=1152, bias=True)                                                                 
          )                                                                                                                                                                                                                                                                           
          (mlp): Qwen3_5MoeVisionMLP(                                                                                                                                                                                                                                                 
            (linear_fc1): Linear(in_features=1152, out_features=4304, bias=True)                                                           
            (linear_fc2): Linear(in_features=4304, out_features=1152, bias=True) 
....
        (56-58): 3 x Qwen3_5MoeDecoderLayer(
          (linear_attn): Qwen3_5MoeGatedDeltaNet(
            (act): SiLUActivation()
            (conv1d): Conv1d(12288, 12288, kernel_size=(4,), stride=(1,), padding=(3,), groups=12288, bias=False)
            (norm): FusedRMSNormGated(128, eps=1e-06, activation=silu)
            (out_proj): FP8Linear(in_features=8192, out_features=4096, bias=False)
            (in_proj_qkv): FP8Linear(in_features=4096, out_features=12288, bias=False)
            (in_proj_z): FP8Linear(in_features=4096, out_features=8192, bias=False)
            (in_proj_b): Linear(in_features=4096, out_features=64, bias=False)
            (in_proj_a): Linear(in_features=4096, out_features=64, bias=False)
          )
          (mlp): Qwen3_5MoeSparseMoeBlock(
            (gate): Qwen3_5MoeTopKRouter()
            (experts): FP8Expert(
              (act_fn): SiLUActivation()
            )
            (shared_expert): Qwen3_5MoeMLP(
              (gate_proj): FP8Linear(in_features=4096, out_features=1024, bias=False)
              (up_proj): FP8Linear(in_features=4096, out_features=1024, bias=False)
              (down_proj): FP8Linear(in_features=1024, out_features=4096, bias=False)
              (act_fn): SiLUActivation()
            )
            (shared_expert_gate): Linear(in_features=4096, out_features=1, bias=False)
          )
          (input_layernorm): Qwen3_5MoeRMSNorm((4096,), eps=1e-06)
          (post_attention_layernorm): Qwen3_5MoeRMSNorm((4096,), eps=1e-06)
        )
        (59): Qwen3_5MoeDecoderLayer(
          (self_attn): Qwen3_5MoeAttention(
            (q_proj): FP8Linear(in_features=4096, out_features=16384, bias=False)
            (k_proj): FP8Linear(in_features=4096, out_features=512, bias=False)
            (v_proj): FP8Linear(in_features=4096, out_features=512, bias=False)
            (o_proj): FP8Linear(in_features=8192, out_features=4096, bias=False)
            (q_norm): Qwen3_5MoeRMSNorm((256,), eps=1e-06)
            (k_norm): Qwen3_5MoeRMSNorm((256,), eps=1e-06)
          )
          (mlp): Qwen3_5MoeSparseMoeBlock(
            (gate): Qwen3_5MoeTopKRouter()
            (experts): FP8Expert(
              (act_fn): SiLUActivation()
            )
            (shared_expert): Qwen3_5MoeMLP(
              (gate_proj): FP8Linear(in_features=4096, out_features=1024, bias=False)
              (up_proj): FP8Linear(in_features=4096, out_features=1024, bias=False)
              (down_proj): FP8Linear(in_features=1024, out_features=4096, bias=False)
              (act_fn): SiLUActivation()
            )
            (shared_expert_gate): Linear(in_features=4096, out_features=1, bias=False)
          )
          (input_layernorm): Qwen3_5MoeRMSNorm((4096,), eps=1e-06)
          (post_attention_layernorm): Qwen3_5MoeRMSNorm((4096,), eps=1e-06)
        )
      )
      (norm): Qwen3_5MoeRMSNorm((4096,), eps=1e-06)
      (rotary_emb): Qwen3_5MoeTextRotaryEmbedding()
    )
  )
  (lm_head): Linear(in_features=4096, out_features=248320, bias=False)
)
[INFO:swift] model_parameter_info: Qwen3_5MoeForConditionalGeneration: 11010.7896M Params (11010.7896M Trainable [100.0000%]), 0.0001M Buffers.

### Additional Information / 补充信息

_No response_

**Comment by Jintao-Huang** [comment]
use transformers 5.8.0
